client/src/main/java/org/apache/rocketmq/client/consumer/MQPullConsumer.java

```diff
@@ -150,6 +149,23 @@ void pullBlockIfNotFound(final MessageQueue mq, final String subExpression, fina
         final int maxNums, final PullCallback pullCallback) throws MQClientException, RemotingException,
         InterruptedException;
 
+    /**
+     * Pulling the messages through callback function,if no message arrival,blocking. Support message selection
+     */
+    void pullBlockIfNotFoundWithMessageSelector(final MessageQueue mq, final MessageSelector selector,
```

Comment:
How about using the `pullBlockIfNotFound` overload directly?

Reply:
The main issue is that when the second parameter is null, it's impossible to distinguish between the two methods.